### PR TITLE
[DONT MERGE] [EARLY FEED BACK]Fix omit assembly version

### PR DESCRIPTION
**Pull request by wli3**

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.NET.Build.Tasks.ConflictResolution;
```
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs]: This file should not reference ConflictResolution, I should move FileUtilities.GetFileVersion to a "higher" name space right?

File: src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+        protected override void ExecuteCore()
+        {
+            GetAssemblyInfoFrom(FilePath);
+        }
+
+        private void GetAssemblyInfoFrom(string FilePath)
+        {
+            FileVersion = FileUtilities.GetFileVersion(FilePath).ToString();
```
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs]: Right API to get FileVersion?

File: src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            var TryGetAssemblyVersion = FileUtilities.TryGetAssemblyVersion(FilePath).ToString();
+            if (TryGetAssemblyVersion != null)
+            {
+                AssemblyVersion = TryGetAssemblyVersion;
```
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs]: And TryGetAssemblyVersion?

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets [hunk 4]
```diff
@@ -411,10 +413,27 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PropertyGroup>
       <_AssemblyInfoFile>$(IntermediateOutputPath)%(_SatelliteAssemblyResourceInputs.Culture)\$(TargetName).resources.cs</_AssemblyInfoFile>
       <_OutputAssembly>$(IntermediateOutputPath)%(_SatelliteAssemblyResourceInputs.Culture)\$(TargetName).resources.dll</_OutputAssembly>
+      <_CoreGenerateSatelliteAssembliesTemplateFile>$(IntermediateOutputPath)$(TargetName)$(TargetExt)</_CoreGenerateSatelliteAssembliesTemplateFile>
     </PropertyGroup>
 
+    <GetAssemblyInfo FilePath="$([System.IO.Path]::GetFullPath('$(_CoreGenerateSatelliteAssembliesTemplateFile)'))" >
```
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: Should I get the full path of the main dll here and pass the full dll path to the task?

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets [hunk 5]
```diff
@@ -411,10 +413,27 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+      <Output TaskParameter="FileVersion"
+            PropertyName="FileVersionFromTemplateFile" />
+      <Output TaskParameter="AssemblyVersion"
+            PropertyName="AssemblyVersionFromTemplateFile" />
+    </GetAssemblyInfo>
+
     <ItemGroup>
       <SatelliteAssemblyAttribute Include="@(AssemblyAttribute)" />
+
+      <SatelliteAssemblyAttribute Include="System.Reflection.AssemblyFileVersionAttribute" Condition="'$(FileVersion)' != ''">
```
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: Should the condition to be '$(GenerateAssemblyVersionAttribute)' == 'false'?

File: test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            var mainAssembliyFileVersioninfo = FileVersionInfo.GetVersionInfo(_mainAssembliyPath);
+            var satelliteAssembliyFileVersioninfo = FileVersionInfo.GetVersionInfo(_satelliteAssembliyPath);
+
+            satelliteAssembliyFileVersioninfo.FileVersion.Should().Be(mainAssembliyFileVersioninfo.FileVersion);
+        }
+
+        [Fact]
+        public void It_should_produce_same_SatelliteAssemblie_AssemblyVersions_as_main()
+        {
+            var mainAssembliy = AssemblyName.GetAssemblyName(_mainAssembliyPath);
```
Comment (wli3) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWanSatelliteAssembliesHaveassemblyVersion.cs]: This is the non-loading API to get assembly name right?

File: src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs (hunk 2 above)
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetAssemblyInfo.cs]: This is OK, but we should get the other parts of FileVersionInfo and convert them to appropriate attribute.

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets (hunk 5 above)
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets]: Always copy from the main assembly no matter what

**Comment by wli3** [comment]
Close to keep working on

**Pull request closed without merging**